### PR TITLE
ci: use binary build stage in python CI

### PR DESCRIPTION
**Pull request by ss2165**

### CODE REVIEW

File: .github/workflows/ci-py.yml [hunk 1]
```diff
@@ -68,8 +71,59 @@ jobs:
...
+        uses: dtolnay/rust-toolchain@stable
+      - name: Build HUGR binary
+        run: cargo build -p hugr-cli
+      - name: Upload the binary to the artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: hugr_binary
+          path: target/debug/hugr
+  test:
+    needs: [changes, check, build_binary]
```
Comment (aborgna-q) [inline .github/workflows/ci-py.yml:97]: No need to wait for the `check` step to run this one.
```suggestion
    needs: [changes, build_binary]
```

**Review by aborgna-q — APPROVED** [review]
LGTM 👍

**Comment by aborgna-q** [comment]
Uhm. Should updates to the rust crate trigger the python tests now?

**Comment by ss2165** [comment]
Ideally only changes to the serialisation should right?

**Comment by aborgna-q** [comment]
Ideally yes.
If we trust the rust-side test to check that we're not breaking the cli tool then we should be OK.

**Comment by ss2165** [comment]
> If we trust the rust-side test to check that we're not breaking the cli tool then we should be OK.

this should be done using @doug-q 's suggestion of a collection of "v1 HUGRs" stored as json that we validate to ensure compatibilty

**Pull request merged**